projects/plugins/jetpack/extensions/extended-blocks/paid-blocks/with-upgrade-banner.jsx

```diff
@@ -70,50 +72,33 @@ export default createHigherOrderComponent(
 			setAttributes( { shouldDisplayFrontendBanner: ! hasParentBanner } );
 		}, [ setAttributes, hasParentBanner ] );
 
-		useEffect( () => {
-			const block = document.querySelector( `.wp-block[data-block="${ clientId }"]` );
-			if ( ! block ) {
-				return;
-			}
-
-			let upgradeBannerContainer = document.querySelector(
-				`.wp-block[data-block="${ clientId }"] > .jetpack-block-upgrade-banner-container`
-			);
-			if ( ! upgradeBannerContainer ) {
-				upgradeBannerContainer = document.createElement( 'div' );
-				upgradeBannerContainer.classList.add( 'jetpack-block-upgrade-banner-container' );
-				block.prepend( upgradeBannerContainer );
-			}
+		const blockType = getBlockType( name );
 
-			render(
-				<UpgradePlanBanner
-					className={ `is-${ name.replace( /\//, '-' ) }-paid-block` }
-					title={ null }
-					align={ attributes?.align }
-					visible={ isBannerVisible }
-					description={ usableBlocksProps?.description }
-					requiredPlan={ requiredPlan }
-					context={ bannerContext }
-					onRedirect={ () => trackUpgradeClickEvent( trackEventData ) }
-				/>,
-				upgradeBannerContainer
-			);
-		}, [
-			attributes?.align,
-			clientId,
-			isBannerVisible,
-			name,
-			requiredPlan,
-			trackEventData,
-			usableBlocksProps?.description,
-		] );
+		/**
+		 * When the api version of the block is 2, it means that blockProps have to be set on the block.
+		 *
+		 * However, in version 1, this wasn't necessary since the entire HOC is wrapped in a block wrapper.
+		 */
+		const blockProps = 2 === blockType.apiVersion ? useBlockProps() : {};
 
 		return (
 			<PaidBlockProvider
 				onBannerVisibilityChange={ setIsVisible }
 				hasParentBanner={ isBannerVisible }
 			>
-				<BlockEdit { ...props } />
+				<div { ...blockProps }>
+					<UpgradePlanBanner
+						className={ `is-${ name.replace( /\//, '-' ) }-paid-block` }
+						title={ null }
+						align={ attributes?.align }
+						visible={ isBannerVisible }
+						description={ usableBlocksProps?.description }
+						requiredPlan={ requiredPlan }
+						context={ bannerContext }
+						onRedirect={ () => trackUpgradeClickEvent( trackEventData ) }
+					/>
+					<BlockEdit { ...props } forceDisableBlockProps={ true } />
```

Comment:
I got mixed feelings about this new `forceDisableBlockProps` prop, since it requires now further work on the edit components of all paid blocks, but this HOC was supposed to abstract all that work.

In other words, previously developers only had to [declare the block as paid](https://github.com/Automattic/jetpack/blob/8a1fee9e2ffebbd2176a457e49d623efcf0ec605/projects/plugins/jetpack/extensions/blocks/payment-buttons/payment-buttons.php#L45-L53) and the upgrade banner would automatically appear. Now, developers are also required to add special logic to the edit component checking this prop.

I recommend trying to find an alternative way to abstract the edit components from the presence of the upgrade banner.

Also, as an aside, I think `forceDisableBlockProps` should be false for v1 blocks.

Reply:
I'm not too fond of using the prop either; the problem is that block API v2 requires a block to have defined on one of its elements the blockProps. With the v1, we don't have this problem because the blocks are automatically wrapped.

Another idea would be to always display the nudges, not just when the block is selected. This way, we won't need to move the toolbar since the nudge will still be visible when the block is not selected - meaning it would be more visible. However, that means having the nudges more invasive and it would affect how we present a payment-block-powered theme for the first time when users install it. 

Having that said, I think this prop should be enough for the current needs; when payment blocks will become a priority, I think the nudges have to be rethought too (e.g. Jetpack upgrade nudges are displayed differently).

Reply:
Actually... using `forceDisableBlockProps` doesn't cover (pun intended) the cover block, since it's a core block 😢